ddtrace/appsec/_utils.py

```diff
@@ -20,6 +21,35 @@
 
 log = get_logger(__name__)
 
+_TRUNC_STRING_LENGTH = 1
+_TRUNC_CONTAINER_DEPTH = 4
+_TRUNC_CONTAINER_SIZE = 2
+
+
+class _observator:
+    def __init__(self):
+        self.string_length: Optional[int] = None
+        self.container_size: Optional[int] = None
+        self.container_depth: Optional[int] = None
+
+    def set_string_length(self, length: int):
```

Comment:
@property and @string_length.set?

Reply:
same thing here, i did not want to add a layer of indirection for something that needs to be as fast as possible.